docs/CLA.md

```diff
@@ -0,0 +1,79 @@
+
+"Submit" means any form of electronic, verbal, or written communication sent to Us or our representatives, including but not limited to electronic mailing lists, source code control systems, and issue tracking systems that are managed by, or on behalf of, Us for the purpose of discussing and improving the Material, but excluding communication that is conspicuously marked or otherwise designated in writing by You as "Not a Contribution."
+
+"Submission Date" means the date on which You Submit a Contribution to Us.
+
+"Effective Date" means the date You execute this Agreement or the date You first Submit a Contribution to Us, whichever is earlier.
+
+"Media" means any portion of a Contribution which is not software.
+
+## 2. Grant of Rights
+
+2.1 Copyright License
+
+(a) You retain ownership of the Copyright in Your Contribution and have the same rights to use or license the Contribution which You would have had without entering into the Agreement.
+
+(b) To the maximum extent permitted by the relevant law, You grant to Us a perpetual, worldwide, non-exclusive, transferable, royalty-free, irrevocable license under the Copyright covering the Contribution, with the right to sublicense such rights through multiple tiers of sublicensees, to reproduce, modify, display, perform and distribute the Contribution as part of the Material; provided that this license is conditioned upon compliance with Section 2.3.
+
+2.2 Patent License
+
+For patent claims including, without limitation, method, process, and apparatus claims which You own, control or have the right to grant, now or in the future, You grant to Us a perpetual, worldwide, non-exclusive, transferable, royalty-free, irrevocable patent license, with the right to sublicense these rights to multiple tiers of sublicensees, to make, have made, use, sell, offer for sale, import and otherwise transfer the Contribution and the Contribution in combination with the Material (and portions of such combination). This license is granted only to the extent that the exercise of the licensed rights infringes such patent claims; and provided that this license is conditioned upon compliance with Section 2.3.
+
+2.3 Outbound License
+
+As a condition on the grant of rights in Sections 2.1 and 2.2, We agree to license the Contribution only under the terms of the license or licenses which We are using on the Submission Date for the Material (including any rights to adopt any future version of a license if permitted).
+
+In addition, We may use the following licenses for Media in the Contribution: Creative Commons Attribution 4.0 International License (including any right to adopt any future version of a license if permitted).
```

Comment:
I added this reference to the CC Attribution License to cover media files that are not covered by the Apache 2.0 software license. Feel free to recommend alternative media licenses.

Reply:
We have third-party copyrighted CAD models in the manufacturing directories. We were given informal permission by McMaster-Carr (by email correspondence) to use those, but they didn't want special mention or anything. Any impact here?

Reply:
IANAL, but my understanding is that they retain copyright (true for all contributors). The essence of the CLA is that contributors are giving us the right to use and distribute the contributed works.  The question, I think, would be do we only have the right to "use", or do we have the right to "use and distribute (per clause 2.1- b)"?

Reply:
They saw our repo and they said what we are doing was ok.